python/ray/tune/tests/test_trial_runner_3.py

```diff
@@ -204,19 +192,18 @@ def testSearchAlgStalled(self):
         runner = TrialRunner(search_alg=search_alg)
         runner.step()
         trials = runner.get_trials()
-        self.assertEqual(trials[0].status, Trial.RUNNING)
+        while trials[0].status != Trial.TERMINATED:
+            runner.step()
 
         runner.step()
-        self.assertEqual(trials[0].status, Trial.TERMINATED)
-
         trials = runner.get_trials()
-        runner.step()
         self.assertEqual(trials[1].status, Trial.RUNNING)
         self.assertEqual(len(searcher.live_trials), 1)
 
         searcher.stall = True
 
-        runner.step()
+        while trials[1].status != Trial.TERMINATED:
+            runner.step()
```

Comment:
for these infinite loops, should we have some sort of function that breaks after X timesteps as a safety check?

i.e.

loop_step(runner, cond=lambda: trials[1].status == Trial.TERMINATED, max_steps=100)

Reply:
Noted. I will keep that in mind.
I want to battle test this change in post submit tests. I foresee there can be some issues here and there(that are not easily showing up with small sample size and local setup). I will monitor the dashboard and will implement something alike if needed.